src/opm/utility/EModel.cpp

```diff
@@ -0,0 +1,474 @@
+/*
+   Copyright 2019 Equinor ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include <opm/utility/EModel.hpp>
+#include <opm/common/utility/FileSystem.hpp>
+
+#include <opm/common/utility/numeric/calculateCellVol.hpp>
+#include <string>
+#include <string.h>
+#include <sstream>
+#include <iterator>
+#include <iomanip>
+#include <algorithm>
+
+
+using EclEntry = std::tuple<std::string, Opm::EclIO::eclArrType, long int>;
+using ParamEntry = std::tuple<std::string, Opm::EclIO::eclArrType>;
+
+
+EModel::EModel(const std::string& filename)
```

Comment:
In order to have value semantics on the initfile you probably need to initialize it an initialiser list:
```
EModel::EModel(const std::string& filename) :
initfile( filename )
{
    ....
}
```